Source/cmGlobalUnixMakefileGenerator3.cxx

```diff
@@ -169,6 +169,46 @@ void cmGlobalUnixMakefileGenerator3::Generate()
     *this->CommandDatabase << "\n]";
     this->CommandDatabase.reset();
   }
+
+    if (this->LinkCommandDatabase) {
+        *this->LinkCommandDatabase << "\n]";
+        this->LinkCommandDatabase.reset();
+    }
+}
+
+void cmGlobalUnixMakefileGenerator3::AddCXXLinkCommand(const std::vector<std::string>& sourceFiles, const std::string& workingDirectory,
+                                                       const std::string& linkCommand)
+{
+  if (!this->LinkCommandDatabase) {
+    std::string commandDatabaseName =
+      this->GetCMakeInstance()->GetHomeOutputDirectory() +
+      "/link_commands.json";
+    this->LinkCommandDatabase =
+      cm::make_unique<cmGeneratedFileStream>(commandDatabaseName);
+    *this->LinkCommandDatabase << "[\n";
+  } else {
+    *this->LinkCommandDatabase << ",\n";
+  }
+  *this->LinkCommandDatabase << "{\n"
+                         << R"(  "directory": ")"
+                         << cmGlobalGenerator::EscapeJSON(workingDirectory)
+                         << "\",\n"
+                         << R"(  "command": ")"
+                         << cmGlobalGenerator::EscapeJSON(linkCommand)
+                         << "\",\n"
+                         << R"(  "files": [)";
```

Comment:
Place newline here, please
```
 "files": ["/root/lapack/build/BLAS/SRC/BLAS/SRC/CMakeFiles/blas.dir/isamax.f.o",
```
This will look better:
```
 "files": [
        "/root/lapack/build/BLAS/SRC/BLAS/SRC/CMakeFiles/blas.dir/isamax.f.o",
```